renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -1493,28 +1493,48 @@ void WrappedVulkan::vkCmdBeginRenderPass(VkCommandBuffer commandBuffer,
     record->MarkResourceFrameReferenced(GetResID(pRenderPassBegin->renderPass), eFrameRef_Read);
 
     VkResourceRecord *fb = GetRecord(pRenderPassBegin->framebuffer);
+    VkResourceRecord *rp = GetRecord(pRenderPassBegin->renderPass);
 
     record->MarkResourceFrameReferenced(fb->GetResourceID(), eFrameRef_Read);
 
     rdcarray<VkImageMemoryBarrier> &barriers = record->cmdInfo->rpbarriers;
 
     barriers.clear();
 
-    if(fb->imageAttachments[0].barrier.sType && fb->imageAttachments[0].record)
+    FramebufferInfo *fbInfo = fb->framebufferInfo;
+    RenderPassInfo *rpInfo = rp->renderPassInfo;
+
+    if(fbInfo->imageAttachments[0].barrier.sType && fbInfo->imageAttachments[0].record)
     {
-      for(size_t i = 0; fb->imageAttachments[i].barrier.sType; i++)
+      bool renderArea_covers_entire_framebuffer =
+          pRenderPassBegin->renderArea.offset.x == 0 && pRenderPassBegin->renderArea.offset.y == 0 &&
+          pRenderPassBegin->renderArea.extent.width >= fbInfo->width &&
+          pRenderPassBegin->renderArea.extent.height >= fbInfo->height && fbInfo->layers == 1;
```

Comment:
Is the check for `fbInfo->layers == 1` necessary here? If we have a framebuffer with multiple layers and a clear loadop, that is still just as good for considering a complete write.